### PR TITLE
[NIFI-12790] Migrate to Angular 17

### PR DESCRIPTION
**Pull request by rfellows**
[NIFI-12790](https://issues.apache.org/jira/browse/NIFI-12790) Migrate to Angular 17

Also migrated to use the new control-flow operators (`@if, @for, @switch, ...`)

### CODE REVIEW

**Comment by joewitt** [comment]
built and used locally from a user (non dev) pov.  Works great.  Overall progress on the new UI is awesome

File: nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/pom.xml [hunk 1]
```diff
@@ -27,7 +27,7 @@
     <version>2.0.0-SNAPSHOT</version>
     <packaging>war</packaging>
     <properties>
-        <node.version>v20.5.1</node.version>
+        <node.version>v21.6.1</node.version>
```
Comment (scottyaslan) [inline nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/pom.xml:30]: FYI:

'''
nifi git:(PR-8410) npm run start

> nifi-web-frontend@0.0.0 start
> ng serve

Node.js version v21.6.1 detected.
Odd numbered Node.js versions will not enter LTS status and should not be used for production. For more information, please see https://nodejs.org/en/about/previous-releases/.
'''

Comment (scottyaslan) [reply nifi-nar-bundles/nifi-framework-bundle/nifi-framework/nifi-web/nifi-web-frontend/pom.xml:30]: https://issues.apache.org/jira/browse/NIFI-12802

**Review by mcgilman — APPROVED** [review]
Thanks for the PR @rfellows! Looks good. Will merge.

**Pull request merged**